tst/EnergyPlus/unit/PlantManager.unit.cc

```diff
@@ -218,4 +222,119 @@ namespace PlantManager {
+
+namespace UserDefinedComponents {
+
+    TEST_F(EnergyPlusFixture, Fix_CoilUserDefined_Test)
+    {
+        bool ErrorsFound = false;
+
+        std::string const idf_objects = delimited_string({
+            "  Coil:UserDefined,",
+            "    CoilUserDef_1, !-Name",
+            "    , !-Overall Model Simulation Program Calling Manager Name",
+            "    Test Program Calling Manager, !-Model Setup and Sizing Program Calling Manager Name",
+            "    2, !-Number of Air Connections",
+            "    Primary_Inlet_Node, !-Air Connection 1 Inlet Node Name",
+            "    Primary_Outlet_Node, !-Air Connection 1 Outlet Node Name",
+            "    Secondary_Inlet_Node, !-Air Connection 2 Inlet Node Name",
+            "    Secondary_Outlet_Node, !-Air Connection 2 Outlet Node Name",
+            "    No, !-Plant Connection is Used",
+            "    , !-Plant Connection Inlet Node Name",
+            "    ; !-Plant Connection Outlet Node Name",
+
+            "  OutdoorAir:Node,",
+            "    Test_OA_Node;",
+
+            "  EnergyManagementSystem:Actuator,",
+            "    TempSetpointLo,          !- Name",
+            "    Test_OA_Node,  !- Actuated Component Unique Name",
+            "    System Node Setpoint,    !- Actuated Component Type",
+            "    Temperature Minimum Setpoint;    !- Actuated Component Control Type",
+
+            "  EnergyManagementSystem:Actuator,",
+            "    TempSetpointHi,          !- Name",
+            "    Test_OA_Node,  !- Actuated Component Unique Name",
+            "    System Node Setpoint,    !- Actuated Component Type",
+            "    Temperature Maximum Setpoint;    !- Actuated Component Control Type",
+
+            "  EnergyManagementSystem:ProgramCallingManager,",
+            "    Test Program Calling Manager,  !- Name",
+            "    BeginNewEnvironment,  !- EnergyPlus Model Calling Point",
+            "    DualSetpointTestControl;  !- Program Name 1",
+
+            "  EnergyManagementSystem:Program,",
+            "    DualSetpointTestControl,",
+            "    Set TempSetpointLo = 16.0,",
+            "    Set TempSetpointHi  = 20.0;",
+
+        });
+
+        ASSERT_TRUE(process_idf(idf_objects));
+        OutAirNodeManager::SetOutAirNodes(*state);
+        EMSManager::CheckIfAnyEMS(*state);
+        state->dataEMSMgr->FinishProcessingUserInput = true;
+
+        bool anyRan;
+        EMSManager::ManageEMS(*state, EMSManager::EMSCallFrom::SetupSimulation, anyRan, ObjexxFCL::Optional_int_const());
+        EMSManager::ManageEMS(*state, EMSManager::EMSCallFrom::BeginNewEnvironment, anyRan, ObjexxFCL::Optional_int_const());
+
+        GetUserDefinedPlantComponents(*state);
+        ASSERT_FALSE(ErrorsFound);
+
+        // Five Node Connections
+        EXPECT_EQ(5, state->dataBranchNodeConnections->NumOfNodeConnections);
+
+        // OA Node
+        EXPECT_EQ(static_cast<int>(NodeInputManager::CompFluidStream::Primary),
```

Comment:
All these casts makes it really tough to read.  It would be much better to use `EXPECT_TRUE(compare_enums(a, b));`  Could you make that change?  Or let me know and I can.

Reply:
Sure, I will make the change.

Reply:
Can you make a shorthand `EXPECT_EQ_ENUM`?

Reply:
I changed the unit test code to use EXPECT_TRUE(compare_enums(a, b)) for the enum comparisons, which works fine. However, still not so sure how to create a new short hand EXPECT_EQ_ENUM() after looking into gtest.cc. Maybe some guidance on how to do it?

Reply:
I agree this would be a nice addition.  But I don't think you need to worry about adding that in this particular PR.  We can make that in a separate PR, and we'll sweep the unit tests and update all of them to use the new macro.